File: app/src/main/java/com/cxh/mvpart/di/FragmentScoped.java
package com.cxh.mvpart.di;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import javax.inject.Scope;

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/3/6
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface FragmentScoped {}

File: app/src/main/java/com/cxh/mvpart/di/FragmentScoped.kt
package com.cxh.mvpart.di

import javax.inject.Scope

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/3/6
 */
@Scope
@kotlin.annotation.Retention
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class FragmentScoped
